homeassistant/components/lyric/sensor.py

```diff
@@ -43,210 +66,122 @@ async def async_setup_entry(
+                        ),
+                        location,
+                        device,
+                    )
+                )
             if device.displayedOutdoorHumidity:
-                cls_list.append(LyricOutdoorHumiditySensor)
-            if device.changeableValues:
-                if device.changeableValues.nextPeriodTime:
-                    cls_list.append(LyricNextPeriodSensor)
-                if device.changeableValues.thermostatSetpointStatus:
-                    cls_list.append(LyricSetpointStatusSensor)
-            for cls in cls_list:
                 entities.append(
-                    cls(
+                    LyricSensor(
                         coordinator,
+                        LyricSensorEntityDescription(
+                            key=f"{device.macID}_outdoor_humidity",
+                            name="Outdoor Humidity",
+                            device_class=DEVICE_CLASS_HUMIDITY,
+                            state_class=STATE_CLASS_MEASUREMENT,
+                            native_unit_of_measurement="%",
+                            value=lambda device: device.displayedOutdoorHumidity,
+                        ),
                         location,
                         device,
-                        hass.config.units.temperature_unit,
                     )
                 )
+            if device.changeableValues:
+                if device.changeableValues.nextPeriodTime:
+                    entities.append(
+                        LyricSensor(
+                            coordinator,
+                            LyricSensorEntityDescription(
+                                key=f"{device.macID}_next_period_time",
+                                name="Next Period Time",
+                                device_class=DEVICE_CLASS_TIMESTAMP,
+                                value=lambda device: get_datetime_from_future_time(
+                                    device.changeableValues.nextPeriodTime
+                                ),
+                            ),
+                            location,
+                            device,
+                        )
+                    )
+                if device.changeableValues.thermostatSetpointStatus:
+                    entities.append(
+                        LyricSensor(
+                            coordinator,
+                            LyricSensorEntityDescription(
+                                key=f"{device.macID}_setpoint_status",
+                                name="Setpoint Status",
+                                icon="mdi:thermostat",
+                                value=lambda device: f"Held until {device.changeableValues.nextPeriodTime}"
+                                if device.changeableValues.thermostatSetpointStatus
+                                == PRESET_HOLD_UNTIL
+                                else LYRIC_SETPOINT_STATUS_NAMES.get(
```

Comment:
inline-if-else is difficult to read. Please define the function above the `entities.append`